src/users/user.schema.js

```diff
@@ -0,0 +1,59 @@
+import mongoose from 'mongoose';
+import mongooseHidden from 'mongoose-hidden';
+import moment from 'moment';
+
+export const UserSchema = new mongoose.Schema(
+  {
+    id: {
+      type: String,
+      required: true,
+      validate: {
+        validator: function(v) {
+          return /^[a-zA-Z0-9]+$/.test(v);
+        },
+        message: props => `${props.value} is not a valid id!`,
+      },
+    },
+    name: { type: String, required: false },
```

Comment:
user.name is not required, which is unusual to me, do you have any concern about that?

Reply:
This system is designed to be used by another system, so I think less constraints are better.
If the `name` is required in business, the other system can add constraint in its own API, and send `name` to this system.